[TECH] Amélioration du code de l'import de sessions en masse (PIX-7119).

AndreiaPena opened the pull request:
## :unicorn: Problème
Le usecase `create-sessions` permettant l'import de session en masse manque de clarté car il gère beaucoup de choses : la validation des données fournies par le csv avec la BDD et la création des sessions et des candidats.

## :robot: Proposition
Transférer la validation des données du usecase dans un nouveau service : `sessions-import-validation-service`

## :rainbow: Remarques
La gestion des erreurs sera traîté dans une autre PR

## :100: Pour tester

P-Jeremy committed:
:sparkles: api: factorize domain session instanciation

P-Jeremy committed:
:sparkles: api: add #isSessionScheduledInThePast to Session model

P-Jeremy committed:
:recycle: api: use #isSessionScheduledInThePast from Session model in usecase

AndreiaPena committed:
sr(api): add function parameters in object

AndreiaPena committed:
:recycle: api: return an error if the file has no session information

P-Jeremy committed:
:recycle: api: use service to check if session is scheduled in the past

alexandrecoin committed:
:recycle: api: use service to check if session has unnecessary info

aceol committed:
:recycle: api:  manage existing session in validation service

P-Jeremy committed:
:sparkles: api: add static method to generate supervisor password

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr5664.review.pix.fr)
  * [App (.org)](https://app-pr5664.review.pix.org)
  * [Orga](https://orga-pr5664.review.pix.fr)
  * [Certif](https://certif-pr5664.review.pix.fr)
  * [Admin](https://admin-pr5664.review.pix.fr)
  * [API](https://api-pr5664.review.pix.fr/api/)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr5664/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr5664/environment)

Thomasevano committed:
:recycle: use service to check if sessions has started

 Co-authored-by: Andreia Pena <andreia.pena@pix.fr>
 Co-authored-by: Quentin Lebouc <quentin.lebouc@pix.fr>
 Co-authored-by: Jérémy Pluquet <jemyplu@gmail.com>
 Co-authored-by: Antoine Ceol <antoine.ceol@gmail.com>
 Co-authored-by: Alexandre Coin <alexandre.pc.coin@gmail.com>

Libouk committed:
:recycle: api: move new session validation and simplify session build

Co-authored-by: Antoine Ceol <antoine.ceol@gmail.com>
Co-authored-by: Jérémy Pluquet <jemyplu@gmail.com>
Co-authored-by: Alexandre Coin <alexandre.coin@pix.fr>

aceol committed:
:recycle: api: fix injection temporairement

l'injection ne se fait pas dans les services, en attendant,
le useCase peut passer les repos au service

P-Jeremy committed:
:recycle: api: remove unnecessary promise

P-Jeremy committed:
:recycle: api: check candidate duplicates in validation service
and refacto check id && session info
